client/http/client.go

```diff
@@ -289,23 +311,70 @@ func (c *client) GetHotReadRegions(ctx context.Context) (*StoreHotPeersInfos, er
 // GetHotWriteRegions gets the hot write region statistics info.
 func (c *client) GetHotWriteRegions(ctx context.Context) (*StoreHotPeersInfos, error) {
 	var hotWriteRegions StoreHotPeersInfos
-	err := c.requestWithRetry(ctx, "GetHotWriteRegions", HotWrite, &hotWriteRegions)
+	err := c.requestWithRetry(ctx,
+		"GetHotWriteRegions", HotWrite,
+		http.MethodGet, nil, &hotWriteRegions)
 	if err != nil {
 		return nil, err
 	}
 	return &hotWriteRegions, nil
 }
 
+// GetRegionStatusByKeyRange gets the region status by key range.
+func (c *client) GetRegionStatusByKeyRange(ctx context.Context, startKey, endKey []byte) (*RegionStats, error) {
+	var regionStats RegionStats
+	err := c.requestWithRetry(ctx,
+		"GetRegionStatusByKeyRange", RegionStatsByKeyRange(startKey, endKey),
+		http.MethodGet, nil, &regionStats,
+	)
+	if err != nil {
+		return nil, err
+	}
+	return &regionStats, nil
+}
+
 // GetStores gets the stores info.
 func (c *client) GetStores(ctx context.Context) (*StoresInfo, error) {
 	var stores StoresInfo
-	err := c.requestWithRetry(ctx, "GetStores", Stores, &stores)
+	err := c.requestWithRetry(ctx,
+		"GetStores", Stores,
+		http.MethodGet, nil, &stores)
 	if err != nil {
 		return nil, err
 	}
 	return &stores, nil
 }
 
+// GetPlacementRulesByGroup gets the placement rules by group.
+func (c *client) GetPlacementRulesByGroup(ctx context.Context, group string) ([]*Rule, error) {
+	var rules []*Rule
+	err := c.requestWithRetry(ctx,
+		"GetPlacementRulesByGroup", PlacementRulesByGroup(group),
+		http.MethodGet, nil, &rules)
+	if err != nil {
+		return nil, err
+	}
+	return rules, nil
+}
+
+// SetPlacementRule sets the placement rule.
+func (c *client) SetPlacementRule(ctx context.Context, rule *Rule) error {
+	ruleJSON, err := json.Marshal(rule)
+	if err != nil {
+		return errors.Trace(err)
```

Comment:
not sure why needs `errors.Trace`?

Reply:
One is to be consistent with the original HTTP call code of TiDB, and the other is that because it is a general library, adding trace information can help us better locate where the call is wrong I think.